go.mod

```diff
@@ -510,11 +511,11 @@ require (
 	k8s.io/csi-translation-lib v0.0.0 // indirect
 	k8s.io/dynamic-resource-allocation v0.0.0 // indirect
 	k8s.io/endpointslice v0.0.0 // indirect
-	k8s.io/gengo v0.0.0-20220902162205-c0856e24416d // indirect
+	k8s.io/gengo v0.0.0-20230829151522-9cce18d56c01 // indirect
 	k8s.io/kms v0.0.0 // indirect
 	k8s.io/kube-aggregator v0.28.6 // indirect
 	k8s.io/kube-controller-manager v0.0.0 // indirect
-	k8s.io/kube-openapi v0.0.0-20230717233707-2695361300d9 // indirect
+	k8s.io/kube-openapi v0.0.0-20230928205116-a78145627833 // indirect
```

Comment:
I suspect you're going to have to pin some of these indirect deps that kube-router wants to bump for you. The versions that kube-router tries to pull in are incompatible with other things embedded in K3s. Check the version of kube-openapi used by this minor version of upstream Kubernetes, and add a `replace` entry to pin it to that version.

https://github.com/kubernetes/kubernetes/blob/release-1.28/go.mod#L118